From e539758510e3f749265b2c7b6199786e0ff7b487 Mon Sep 17 00:00:00 2001
From: yewenhao <10165101137@stu.ecnu.edu.cn>
Date: Sun, 23 May 2021 11:56:37 +0800
Subject: [PATCH 5/9] add and update op CSEC

---
 myFlaskMain.py |  7 ++-----
 myWebUtil.py   | 41 +++++++++++++++++++----------------------
 2 files changed, 21 insertions(+), 27 deletions(-)

diff --git a/myFlaskMain.py b/myFlaskMain.py
index eefc1d2..c5c33d8 100644
--- a/myFlaskMain.py
+++ b/myFlaskMain.py
@@ -20,15 +20,12 @@ def DBOperation():
         op = request.form['op']
         print("op=%s" % op)
         if op == "add":
-            myWebUtil.InsertNewWebData(db,cursor,myWebUtil.Request2JSon(request))
+            myWebUtil.HandleNewWant(db, cursor, request)
         elif op == "delete":
             webdataid = request.form['id']
             myWebUtil.DeleteWebDataById(db,cursor,webdataid)
         elif op == "update":
-            webdataid = request.form['id']
-            old = myWebUtil.SelectOneWebDataById(cursor,webdataid)
-            new = myWebUtil.Request2JSon(request,myWebUtil.JSon2WebData(old['content']))
-            myWebUtil.DBUpdateOneWebData(db,cursor,new,webdataid)
+            myWebUtil.HandleGenralUpdate(db, cursor, request)
         elif op == "config pattern":
             myWebUtil.HandleConfigPattern(request)
         elif op == "config dir":
diff --git a/myWebUtil.py b/myWebUtil.py
index 344aa1c..ea43b48 100644
--- a/myWebUtil.py
+++ b/myWebUtil.py
@@ -133,7 +133,7 @@ def DeleteWebDataById(db,cursor,ID):
     db.commit()
     print("delete id=%s"%ID)
 
-def InsertNewWebData(db,cursor,content):
+def DBInsertNewWebData(db,cursor,content):
     sql = "insert into webdata(content) values (%s)"
     cursor.execute(sql,content)
     db.commit()
@@ -215,33 +215,30 @@ def DelParentList(db, cursor, ID, parentID):
 '''
 Handle Op
 '''
-def Request2JSonWant(request,old):
+def HandleNewWant(db, cursor, request):
     new = {}
-    print(request.form)
+    old = {}
     for tag in jsonTags:
         if tag in request.form:
             new[tag] = request.form[tag]
-    if request.form['op'] == 'add':
-        new['status'] = 'todo'
-        new['originalDate'] = time.localtime()
-    elif request.form['op'] == 'update':
-        for tag in old:
-            if tag not in new:
-                new[tag] = old[tag]
+    new['status'] = 'todo'
+    new['originalDate'] = time.localtime()
     new['updateDate'] = time.localtime()
     new['history'] = BuildHistory(old, new)
-    return json.dumps(new)
-
-def Request2JSon(request,old = {}):
-    data = {}
-    # if 'type' not in request.form:
-        # print("no type in request.form")
-        # return json.dumps(data)
-    if ('type' in old and old['type'] == 'want') or ('type' in request.form and request.form['type'] == 'want'):
-        return Request2JSonWant(request,old)
-    else:
-        print("type %s not supported"%request.form['type'])
-    return json.dumps(data)
+    content = WebData2JSon(new)
+    DBInsertNewWebData(db, cursor, content)
+
+def HandleGenralUpdate(db, cursor, request):
+    ID = request.form['id']
+    old = JSon2WebData(SelectOneWebDataById(cursor,ID)['content'])
+    new = {}
+    for tag in jsonTags:
+        if tag in request.form:
+            new[tag] = request.form[tag]
+    for tag in old:
+        if tag not in new:
+            new[tag] = old[tag]
+    UpdateOneWebDataCommon(db, cursor, ID, old, new)
 
 def HandleAddSubList(db, cursor, request):
     ID = request.form['id']
-- 
2.39.2 (Apple Git-143)

