From fda8eeb990fc9a365cff0b1b421b4ddb4da86ec1 Mon Sep 17 00:00:00 2001
From: yewenhao <10165101137@stu.ecnu.edu.cn>
Date: Mon, 7 Jun 2021 22:39:53 +0800
Subject: [PATCH 8/9] #73:allow adding parentID when creating new want

---
 myWebUtil.py             | 12 +++++++++++-
 templates/wantanddo.html | 12 ++++++------
 2 files changed, 17 insertions(+), 7 deletions(-)

diff --git a/myWebUtil.py b/myWebUtil.py
index 4bbbc4c..279ffe5 100644
--- a/myWebUtil.py
+++ b/myWebUtil.py
@@ -168,6 +168,12 @@ def UpdateOneWebDataCommon(db, cursor, ID, old, new):
     content = WebData2JSon(new)
     DBUpdateOneWebData(db, cursor, content, ID)
 
+def DBGetMaxID(db, cursor):
+    cursor.execute("select max(id) as id from webdata")
+    result = cursor.fetchone()
+    maxID = result['id']
+    return maxID
+    
 '''
 blockList[] subList[] parentList[]
 '''
@@ -225,12 +231,16 @@ def HandleNewWant(db, cursor, request):
     for tag in defaultJsonTagsValue:
         if tag not in new:
             new[tag] = defaultJsonTagsValue[tag]
-    # new['status'] = 'todo'
     new['originalDate'] = time.localtime()
     new['updateDate'] = time.localtime()
     new['history'] = BuildHistory(old, new)
     content = WebData2JSon(new)
     DBInsertNewWebData(db, cursor, content)
+    if 'parentID' in request.form and not request.form['parentID'] is None:
+        newID = DBGetMaxID(db, cursor)
+        parentID = int(request.form['parentID'])
+        AddParentList(db, cursor, newID, parentID)
+        AddSubList(db, cursor, parentID, newID)
 
 def HandleGenralUpdate(db, cursor, request):
     ID = request.form['id']
diff --git a/templates/wantanddo.html b/templates/wantanddo.html
index f8ba97a..3e9a932 100644
--- a/templates/wantanddo.html
+++ b/templates/wantanddo.html
@@ -23,12 +23,12 @@
 			  <option type="number" value="2 Median" selected="selected">Median</option>
 			  <option type="number" value="3 Low">Low</option>
 			  </select>
-			  <!-- parent= -->
-			  <!-- {% if config['currentDir'] == -1 %} -->
-			  <!-- <input type="number" name="parent"> -->
-			  <!-- {% else %} -->
-			  <!-- <input type="number" name="parent" value="{{config['currentDir']}}"> -->
-			  <!-- {% endif %} -->
+			  parent=
+			  {% if config['currentDir'] == -1 %}
+			  <input type="number" name="parentID">
+			  {% else %}
+			  <input type="number" name="parentID" value="{{config['currentDir']}}">
+			  {% endif %}
               <p><button type="add">add</button></p>
         </form>
 		<hr>
-- 
2.39.2 (Apple Git-143)

